packages/flutter/lib/src/material/page.dart

```diff
@@ -115,23 +115,26 @@ class _CupertinoBackGestureController extends NavigationGestureController {
   }) : super(navigator);
 
   AnimationController controller;
-  VoidCallback onDisposed;
+  final VoidCallback onDisposed;
 
   @override
   void dispose() {
-    super.dispose();
-    onDisposed();
     controller.removeStatusListener(handleStatusChanged);
     controller = null;
+    onDisposed();
+    super.dispose();
   }
 
   @override
   void dragUpdate(double delta) {
-    controller.value -= delta;
+    controller?.value -= delta;
```

Comment:
I can turn these into asserts if you like.  They're possible to trigger if you do wacky things with your tree, but maybe we should say that's an error.